lib/UnexpectedExpressMocker.js

```diff
@@ -0,0 +1,366 @@
+const makePromise = require('unexpected/lib/makePromise');
```

Comment:
Could you use `require('unexpected').promise` here instead? It's part of the public api and should do the same. I realize that this is an intermediate state and we probably want this to be a separate module that doesn't rely on unexpected's promises.